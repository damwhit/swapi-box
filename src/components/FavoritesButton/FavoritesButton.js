import PropTypes from 'prop-types';
import React from 'react';
import { NavLink } from 'react-router-dom';

function FavoritesButton({ onClick, numFavorites }) {
  return (
    <NavLink to="/favorites" className="link--favorite">
      <button
        className="btn--favorite"
        onClick={onClick}
        type="button"
      >
        Show Favorites (
        {numFavorites}
        )
      </button>
    </NavLink>
  );
}

FavoritesButton.propTypes = {
  onClick: PropTypes.func.isRequired,
  numFavorites: PropTypes.number.isRequired,
};

export default FavoritesButton;
