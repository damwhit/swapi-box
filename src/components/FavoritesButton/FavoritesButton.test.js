import React from 'react';
import { shallow } from 'enzyme';
import FavoritesButton from './FavoritesButton';

describe('FavoritesButton', () => {
  const onClickMock = jest.fn();
  const wrapper = shallow(
    <FavoritesButton
      onClick={onClickMock}
      numFavorites={3}
    />
  );
  

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call the onClick prop when clicked', () => {
    const button = wrapper.find('.btn--favorite');
    button.simulate('click');
    expect(onClickMock).toHaveBeenCalled();
  });
});
