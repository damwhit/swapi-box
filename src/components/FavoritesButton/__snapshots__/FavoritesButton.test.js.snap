// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesButton should match the snapshot with all data passed in correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FavoritesButton
    numFavorites={3}
    onClick={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "activeClassName": "active",
      "aria-current": "page",
      "children": <button
        className="btn--favorite"
        onClick={[MockFunction]}
        type="button"
      >
        Show Favorites (
        3
        )
      </button>,
      "className": "link--favorite",
      "to": "/favorites",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Show Favorites (",
          3,
          ")",
        ],
        "className": "btn--favorite",
        "onClick": [MockFunction],
        "type": "button",
      },
      "ref": null,
      "rendered": Array [
        "Show Favorites (",
        3,
        ")",
      ],
      "type": "button",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "activeClassName": "active",
        "aria-current": "page",
        "children": <button
          className="btn--favorite"
          onClick={[MockFunction]}
          type="button"
        >
          Show Favorites (
          3
          )
        </button>,
        "className": "link--favorite",
        "to": "/favorites",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Show Favorites (",
            3,
            ")",
          ],
          "className": "btn--favorite",
          "onClick": [MockFunction],
          "type": "button",
        },
        "ref": null,
        "rendered": Array [
          "Show Favorites (",
          3,
          ")",
        ],
        "type": "button",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
